sled-agent/src/long_running_tasks.rs

```diff
@@ -46,6 +46,10 @@ pub struct LongRunningTaskHandles {
     /// for establishing zpools on disks and managing their datasets.
     pub storage_manager: StorageHandle,
 
+    /// A mehcanism for talking to the [`StorageMonitor`], which reacts to disk
+    /// changes and updates the dump devices.
+    pub storage_monitor_handle: StorageMonitorHandle,
```

Comment:
This info is plumbed as a bridge between the `StorageMonitor` and the `SledAgent`, so the sled agent can inspect "what version of disks has the `StorageMonitor` acted upon last?"